tests/numerics/ad/test_operators.py

```diff
@@ -651,6 +629,170 @@ def _compare_ad_objects(a, b):
 
 
+def test_ad_variable_prev_iter():
+    # Test only 1 variable, the rest should be covered by other tests
+    mdg, _ = pp.mdg_library.square_with_orthogonal_fractures(
+        "cartesian",
+        {"cell_size": 0.2},
+        fracture_indices=[1],
+    )
+    eqs = pp.ad.EquationSystem(mdg)
+
+    # Integer to test the depth of prev _*, could be a test parameter, but no need
+    depth = 5
+    var_name = 'foo'
+    vec = np.ones(mdg.num_subdomain_cells())
+
+    eqs.create_variables(
+        var_name, dof_info={"cells": 1}, subdomains=mdg.subdomains()
+    )
+    var = eqs.md_variable(var_name)
+
+    # Starting point is time step index is None, iterate index is 0
+    # (current time and iter)
+    assert var.time_step_index is None
+    assert var.iterate_index == 0 
+
+    # Set values except for the last step
+    for i in range(depth - 1):
+        eqs.set_variable_values(vec * i, [var], iterate_index=i)
+
+    # prohibit prev iter variable to also be prev time
+    with pytest.raises(ValueError):
+        var_pi = var.previous_iteration()
+        _ = var_pi.previous_timestep()
+
+    # Evaluating the last step, should raise a key error because no values set
+    with pytest.raises(KeyError):
+        # different steps, because iter starts at 0, time at -1
+        var_prev = var.previous_iteration(steps = depth - 1) 
+        _ = var_prev.value(eqs)
+
+    # Evaluate prev iter and check that the values are what they're supposed to be
+    for i in range(1, depth - 1):
+        var_i = var.previous_iteration(steps = i)
+        val_i = var_i.value(eqs)
+        assert np.allclose(val_i, i)
+
+        # prev iter has no Jacobian
+        ad_i = var_i.value_and_jacobian(eqs)
+        assert np.all(ad_i.jac.A == 0.)
+
+    # Test creating with explicit stepping and recursive stepping
+    vars_exp = [var.previous_iteration(steps=i) for i in range(1, depth - 1)]
+
+    vars_rec = []
+    for i in range(1, depth - 1):
+        var_i = copy.copy(var)
+        for _ in range(i):
+            var_i = var_i.previous_iteration()
+        vars_rec.append(var_i)
+
+    assert len(vars_exp) == len(vars_rec)
+    vals_exp = [v.value(eqs) for v in vars_exp]
+    vals_rec = [v.value(eqs) for v in vars_rec]
+
+    for v_e, v_r in zip(vals_exp, vals_rec):
+        assert np.allclose(v_e, v_r)
+
+    # Testing IDs. NOTE as of now, variables at prev iter have the same ID until
+    # full support is given
+    all_ids = set([var.id] + [v.id for v in vars_exp] + [v.id for v in vars_rec])
+    assert len(all_ids) == 1
+
+    # Testing index values. Should be increasing with depth, while time step index stays
+    # None (current time)
+    for i in range(1, depth - 1):
+        assert vars_exp[i - 1].iterate_index == i
+        assert vars_exp[i - 1].time_step_index is None
+        assert vars_rec[i - 1].iterate_index == i
+        assert vars_rec[i - 1].time_step_index is None
+
+
+def test_ad_variable_prev_time():
```

Comment:
Testing of arbitrary prev time depth.
Indexation logic is different (prev time start with index 0, where as prev iter starts with index 1.
Otherwise the code is quite similar to prev iter test.

Reply:
Does this mean we can unify the test if we unify the index scheme?

Reply:
yes, we can unify the test and make 1 parametrized test, after unifying the index logic.

Reply:
My comment regarding pragmatism when polishing code (posted somewhere in operators.py) applies here: Feel free to postpone this unification in the interest of efficiency.